### PR TITLE
PTX-3055 Add support for Pure FB backend

### PR DESCRIPTION
**Pull request by nikolaypopov**
Add support for Pure FB backend for shared and sharedv4 type volumes.

This is needed for https://portworx.atlassian.net/browse/PTX-3220

### CODE REVIEW

File: drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: wordpress-sc-fb-nfs4
+provisioner: kubernetes.io/portworx-volume
+parameters:
+  backend: "pure_file"
+mountOptions:
+- nfsvers=4.1
```
Comment (adityadani) [inline drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml:9]: @pure-jliao Is this always needed? Do we need to test for other versions as well ?

**Review by adityadani — APPROVED** [review]

File: drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml (hunk 1 above)
Comment (pure-jliao) [reply drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml:9]: right now only support `nfsvers=3` and `nfsvers=4.1`, we could leave `mountOptions` empty to use v3 by default.

File: deployments/deploy-ssh.sh [hunk 2]
```diff
@@ -415,6 +419,7 @@ spec:
             "--storage-upgrade-endpoint-version=$UPGRADE_ENDPOINT_VERSION",
             "--enable-stork-upgrade=$ENABLE_STORK_UPGRADE",
             "--secret-type=$SECRET_TYPE",
+            "--flash-blade-proxy-volumes=$IS_FLASH_BLADE_PROXY_VOLUMES",
```
Comment (pure-jliao) [inline deployments/deploy-ssh.sh:422]: one suggestion, in the future we will support FA proxy volume as well, and our nsm lib is able to arrange request to either FA or FB based on the storage class, so probably a flag like `pure-volumes` might be better?

**Review by pure-jliao — APPROVED** [review]
looks good, only one suggestion regarding the flag naming.

File: deployments/deploy-ssh.sh (hunk 2 above)
Comment (nikolaypopov) [reply deployments/deploy-ssh.sh]: Done

File: drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml (hunk 1 above)
Comment (nikolaypopov) [reply drivers/scheduler/k8s/specs/wordpress-fb-nfs4/storage.yaml:9]: This spec is specifically called `wordpress-fb-nfs4`, so its ok to leave the mount option, the default `wordpress` spec will use the default nfs3 as there is no `mountOptions` specified.

**Pull request merged**